DuckDuckGo/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachine.swift

```diff
@@ -136,7 +151,7 @@ final class ContextualOnboardingStateMachine: ContextualOnboardingDialogTypeProv
 
     private weak var lastVisitTab: Tab?
     private var lastVisitSite: URL?
-    private var notBlockedTrackerSeen: Bool = false
+    var notBlockedTrackerSeen: Bool = false
```

Comment:
not sure how I feel about having this exposed only for testing purpose.

By any chance, could we call `updateStateFor(tab:)` and let the state machine set that without exposing it?